### PR TITLE
[FLINK-6488] [scripts] Mark deprecated 'start-local.sh' and 'stop-local.sh' scripts

### PR DESCRIPTION
**Pull request by zhangminglei**
Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [ ] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [ ] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [ ] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

**Comment by greghogan** [comment]
@zhangminglei, the ticket also lists having `start-local.sh` call `start-cluster.sh` rather than starting a JobManager in local mode.

**Comment by zhangminglei** [comment]
@greghogan Thanks. I just forgot update the title, it is not remove but deprecated instead. And I would suggest you can take a look at this jira https://issues.apache.org/jira/browse/FLINK-6488.

**Comment by zentol** [comment]
Greg is correct, the start-local.sh script should still call the start-cluster.sh script.

Here's what i wrote in the JIRA:
> We will change start-local.sh to a) print a deprecation warning and b) call start-cluster.sh.

**Comment by zhangminglei** [comment]
@zentol Ooooops. I was wrong. I also did the same thing in ```stop-local.sh```

**Comment by zentol** [comment]
You are changing the permissions for the `start-local.sh` script.

File: flink-dist/src/main/flink-bin/bin/start-local.sh [hunk 1]
```diff
@@ -19,10 +19,12 @@
...
 STREAMING_MODE=$1
 
+echo "Warning: this file is deprecated and will be remove in the future"
+
 bin=`dirname "$0"`
 bin=`cd "$bin"; pwd`
 
 . "$bin"/config.sh
 
 # local mode, only bring up job manager. The job manager will start an internal task manager
```
Comment (zentol) [inline flink-dist/src/main/flink-bin/bin/start-local.sh]: comment can be removed

File: flink-dist/src/main/flink-bin/bin/stop-local.sh [hunk 2]
```diff
@@ -17,11 +17,12 @@
...
 ################################################################################
 
+echo "Warning: this file is deprecated and will be remove in the future"
 
 bin=`dirname "$0"`
 bin=`cd "$bin"; pwd`
 
 . "$bin"/config.sh
 
 # stop local job manager (has an internal task manager)
```
Comment (zentol) [inline flink-dist/src/main/flink-bin/bin/stop-local.sh]: comment can be removed

File: flink-dist/src/main/flink-bin/bin/start-local.sh [hunk 3]
```diff
@@ -19,10 +19,12 @@
 
 STREAMING_MODE=$1
 
+echo "Warning: this file is deprecated and will be remove in the future"
```
Comment (zentol) [inline flink-dist/src/main/flink-bin/bin/start-local.sh]: We can be specific and say removed in 1.5, also missing period at the end.

File: flink-dist/src/main/flink-bin/bin/stop-local.sh [hunk 4]
```diff
@@ -17,11 +17,12 @@
 # limitations under the License.
 ################################################################################
 
+echo "Warning: this file is deprecated and will be remove in the future"
```
Comment (zentol) [inline flink-dist/src/main/flink-bin/bin/stop-local.sh]: We can be specific and say removed in 1.5, also missing period at the end.

File: flink-dist/src/main/flink-bin/bin/start-local.sh (hunk 3 above)
Comment (zentol) [inline flink-dist/src/main/flink-bin/bin/start-local.sh]: typo: remove -> removeda

File: flink-dist/src/main/flink-bin/bin/stop-local.sh (hunk 4 above)
Comment (zentol) [inline flink-dist/src/main/flink-bin/bin/stop-local.sh]: typo: remove -> removeda

**Comment by zhangminglei** [comment]
@zentol Thanks a bunch for your review. I have updated the code.

**Comment by zentol** [comment]
you're still modifying the file permissions of the `start-local.sh` file.

**Comment by zhangminglei** [comment]
Sorry, I dont know what you exactly mean. I just modified the scripts. Why did I modify the permissions of this file ? I really need your help. Thanks.

**Comment by zhangminglei** [comment]
It seems that I didnt do anything relevant to the file permissions. Does not make sense.

**Comment by zentol** [comment]
Under the "Files changed" tab on GitHub you can see this ` 100755 → 100644 ` next to the `start-local.sh` script, which means that you removed (somehow) the execute permission from this file.

**Comment by zhangminglei** [comment]
@zentol OMG. I see now. OKay, I will take a look. Thanks a bunch.

**Comment by zhangminglei** [comment]
@zentol It seems weird why this stuff happened. I have updated the code though. It looks good now.

File: flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala [hunk 5]
```diff
@@ -472,7 +472,7 @@ class JobManager(
           taskManagerGateway match {
             case x: ActorTaskManagerGateway =>
               handleTaskManagerTerminated(x.getActorGateway().actor(), instance.getId)
-            case _ => log.debug(s"Cannot remove reosurce ${resourceID}, because there is " +
+            case _ => log.debug(s"Cannot remove resource ${resourceID}, because there is " +
```
Comment (zentol) [inline flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala]: Can you move this change into a separate commit?

Comment (zhangminglei) [inline flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala]: OKay. I'd love to do this. I will push again soon.

Comment (zhangminglei) [inline flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala]: I think I have finished that. Please check it. :XD

**Comment by zentol** [comment]
+1 to merge.

**Comment by zentol** [comment]
merging.

**Comment by zhangminglei** [comment]
Also say thanks to you ~ :XD
